Assets/Scripts/ImprovedCharSelection.cs

```diff
@@ -0,0 +1,124 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+// just to see the profile
+public class ImprovedCharSelection : MonoBehaviour
+{
+    private GameObject[,] characterList;
+    private int classIndex = 0;
+    private int colorIndex = -1;
+
+    private void Start()
+    {
+        classIndex = PlayerPrefs.GetInt("ClassSelected");
+        colorIndex = PlayerPrefs.GetInt("ColorSelected");
```

Comment:
Could `colorIndex` be set to 0 when `ColorSelected` is not found in `PlayerPrefs`?
Also we should use `PlayerPrefs` on the server. @antonioganea right?